### ISSUE TITLE
It doesn't work on Android 6.0

### ISSUE THREAD

**Issue by lennonpetrick** [ArthurHub/Android-Image-Cropper#181]
Hi there. 
I tested on Android prior to 6.0 which worked perfectly. However, on Android 6.0 It can't be opened. Do I have to do any config or something?

**Comment by ArthurHub** [comment]
Probably permissions, see [this wiki page](https://github.com/ArthurHub/Android-Image-Cropper/wiki/Pick-image-for-cropping-from-Camera-or-Gallery)

**Comment by lennonpetrick** [comment]
I already did it, but it didn't work. I put read and write permissions into manifest and also into java class like that wiki page, but neither the permission dialog nor the crop app have shown up

**Comment by ArthurHub** [comment]
Try running the sample app in the library, if it doesn't work there then it's a bug, otherwise:
 1. logcat should have the error or error returned from the library API.
 2. Compare your code to the code in the sample app.
 3. Provide more info on what doesn't work and sample of your code.

**Comment by lennonpetrick** [comment]
I've run the sample app and it works perfectly on Android 6.0. However, the sample uses the `CropImageView` component which is different of my approach.

What I'm doing is to call the CropImageActivity for cropping the pic to me like the Readme.md explains such as `CropImage.activity(imageUri).start(getContext(), this);`. 
That works on Android prior to 6.0, but on Android 6.0 gives the error `pickImageResult.jpeg: open failed: ENOENT (No such file or directory)`

The code below is like I do:

```
@Override
    @SuppressLint("NewApi")
    public void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);

        if (requestCode == REQUEST_CAMERA_APP && resultCode == RESULT_OK) {

            mCropImageURI = CropImage.getPickImageResultUri(getContext(), data);

            if (CropImage.isReadExternalStoragePermissionsRequired(getContext(), mCropImageURI)) {

                requestPermissions(new String[]{Manifest.permission.READ_EXTERNAL_STORAGE},   CropImage.PICK_IMAGE_PERMISSIONS_REQUEST_CODE);

            }
            else {

                CropImage.activity(mCropImageURI).start(getContext(), PhotoConfirmationFragment.this);

            }

        }
        else if (requestCode == CropImage.CROP_IMAGE_ACTIVITY_REQUEST_CODE && resultCode == RESULT_OK) {

            mImgPhotoTaken.setImageURI(CropImage.getActivityResult(data).getUri());

            showButtons(true);

        }

    }

    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String permissions[], @NonNull int[] grantResults) {

        if (requestCode == CropImage.PICK_IMAGE_PERMISSIONS_REQUEST_CODE) {

            if (mCropImageURI != null && grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {

                CropImage.activity(mCropImageURI).start(getContext(), PhotoConfirmationFragment.this);

            }
            else {

                mIActivityController.showSnackBar(new Message(null, getString(R.string.warning_permission_required), Message.INFO));

            }

        }

    }
```

**Comment by ArthurHub** [comment]
The library has two sample apps, the second one is for testing CropActivity, check it out.

It definitely looks like permissions issue, maybe you need `WRITE_EXTERNAL_STORAGE`...
Compare your code to the CropActivity sample app, it is much simpler and read [Pick image for cropping from Camera or Gallery](https://github.com/ArthurHub/Android-Image-Cropper/wiki/Pick-image-for-cropping-from-Camera-or-Gallery) wiki page.

**Comment by ben-j69** [comment]
I have no time to fix your issue but it work with no problem on Android with the correct permissions management.

**Comment by lennonpetrick** [comment]
Actually I have found the solution out. The issue is `CropImage.getPickImageResultUri(getContext(), data)` returns a different Uri path which is not the same as Camera app returns. So I had to give the exactly Uri path to Camera app so that it could store the picture at right place. By the way, I have created a method to deal with it into the code and sent a pull request, but it was not approved yet.

**Comment by ArthurHub** [comment]
I have merger you PR, though I must say both paths set the output URI in the same way. why would the existing `getCameraIntents` act differently?

**Comment by lennonpetrick** [comment]
No, the `getCameraIntents` set a default Uri using `getCaptureImageOutputUri(context)`. The another method the users can specify which Uri they want, so that they get their picture at the specific path. However, if the Uri is passed null, then the code uses the default path.

**Comment by ArthurHub** [comment]
Oh, I see, thx.

**Comment by ArthurHub** [comment]
available in 2.4.0

**Comment by jagadishlakkur** [comment]
dependency 2.7.0 is not accepting and i also if i try with 2.4.0 its ask Uri as a parameter to activity() method. how to get uri?
